### DISCUSSION TITLE
Impossible to specify both name and input_name

### DISCUSSION THREAD

**Discussion by dspicher** [async-graphql/async-graphql#985] [Q&A]
I am trying to derive both `SimpleObject` and `InputObject` on a struct, and would like to rename both objects to something different than the struct name.

Looking at the [source](https://github.com/async-graphql/async-graphql/blob/2a6abe7cae481595e267414a0a0a7a9f1f808d25/derive/src/input_object.rs#L43-L47), this seems impossible, as `name` is given preference over `input_name` for the `InputObject`, leading to a name collision. I would have expected this to be the other way around.

Do you consider this a bug or a feature? :)

**Comment by toxeus** [comment]
@dspicher there is a test that asserts for your desired behavior

https://github.com/async-graphql/async-graphql/blob/2a6abe7cae481595e267414a0a0a7a9f1f808d25/tests/input_object.rs#L353-L408

**Comment by dspicher** [comment]
That is not the desired behaviour, I want to rename the object both as simple and input object.

**Reply by toxeus** [reply]
Ok, so you are also using a `#[graphql(name = "MyRenamedObject")]` attribute compared to the test. 
Interestingly, the `input_name` attribute is not even documented: https://docs.rs/async-graphql/latest/async_graphql/derive.InputObject.html#macro-attributes
To me that looks like a bug.

**Reply by sunli829** [reply]
> 

You can't do that, Object and InputObject must have different names.

**Reply by toxeus** [reply]
But they have different names assuming both attributes are used and the InputObject would prefer `input_name` over `name`. So in the test above:
```rust
     #[derive(SimpleObject, InputObject)] 
     #[graphql(name = "MyRenamedObject")] 
     #[graphql(input_name = "MyObjectInput")] 
     #[allow(dead_code)] 
     struct MyObject { 
```

Currently, this doesn't work because InputObject would ignore `input_name` and pick `name` instead.

**Comment by sunli829** [comment]
Sorry, it's a bug! 😂

**Reply by toxeus** [reply]
:+1:
